### ISSUE TITLE
Test

### ISSUE THREAD

**Issue by salvino080-coder** [pytorch/xla#9740]
"""
Smoke E2E test for PyTorch/XLA: run a short training on XLA device (or CPU fallback).

Run with:
  pytest tests/e2e/test_small_model.py -q
"""
import torch
import torch.nn as nn
import torch.optim as optim

try:
    import torch_xla.core.xla_model as xm
    from torch_xla.distributed.parallel_loader import MpDeviceLoader
except Exception:
    xm = None
    MpDeviceLoader = None

def get_device():
    return xm.xla_device() if xm is not None else torch.device("cpu")

class TinyNet(nn.Module):
    def __init__(self):
        super().__init__()
        self.net = nn.Sequential(
            nn.Flatten(),
            nn.Linear(3*32*32, 128),
            nn.ReLU(),
            nn.Linear(128, 10)
        )
    def forward(self, x):
        return self.net(x)

def synthetic_dataset(batch_size, steps, device):
    for _ in range(steps):
        x = torch.randn(batch_size, 3, 32, 32).to(device)
        y = torch.randint(0, 10, (batch_size,)).to(device)
        yield x, y

def test_small_train_runs():
    device = get_device()
    model = TinyNet().to(device)
    opt = optim.SGD(model.parameters(), lr=0.01)
    criterion = nn.CrossEntropyLoss()

    steps = 10
    batch_size = 32
    loader = synthetic_dataset(batch_size, steps, device)

    losses = []
    for xb, yb in loader:
        opt.zero_grad()
        out = model(xb)
        loss = criterion(out, yb)
        loss.backward()
        if xm is not None:
            xm.optimizer_step(opt)
            xm.mark_step()
        else:
            opt.step()
        losses.append(loss.item())
    assert len(losses) == steps
    # sanity: loss should be finite
    assert all((not torch.isnan(torch.tensor(l)).item()) for l in losses)
